### PR TITLE
Patterns-II-HOCs

### PR DESCRIPTION
**Pull request by renzowolox**
## Summary
* HOCs added

## Trello Card
* https://trello.com/c/cHVNDyMb

### CODE REVIEW

File: react/tic-tac-toe/src/app/components/AuthHOC.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { compose } from 'recompose';
+import Spinner from 'react-spinkit';
+
+const withIsLogged = (isLogged) => (Component) => (props) => {
+  console.log(props);
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/app/components/AuthHOC.js]: Remove it :smile:

File: react/tic-tac-toe/src/app/index.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import Game from './screens/Game';
 import Login from './screens/Login';
 import NotFound from './screens/NotFound';
 import Profile from './screens/Profile';
+import { withConditionalRendering } from './components/AuthHOC';
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/app/index.js]: `AuthHoc` :stuck_out_tongue:

File: react/tic-tac-toe/src/app/index.js [hunk 3]
```diff
@@ -14,7 +15,7 @@ function App() {
   return (
     <Switch>
       <Route path={ROUTES.login} component={Login} />
-      <AuthorizedRoute path={ROUTES.game} component={Game} />
+      <AuthorizedRoute path={ROUTES.game} component={withConditionalRendering(Game)} />
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/app/index.js]: You can add directly into the `Game` component and also to `Profile`

File: react/tic-tac-toe/src/app/components/AuthHOC.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  return isLogged(props) 
+    ? <Component { ...props } />
+    : <Spinner name='circle' />
+}
+
+const isLogged = (props) => props.isLogged;
```
Comment (fededirocco34) [inline react/tic-tac-toe/src/app/components/AuthHOC.js]: Is it necessary? Try `export default withIsLogged`

**Review by ghost — APPROVED** [review]

**Review by erikankodelia — APPROVED** [review]

**Pull request merged**